jobs/migrate_indexes.go

```diff
@@ -0,0 +1,108 @@
+	logger := utils.LoggerFromContext(ctx)
+
+	exec := usecases.NewExecutorFactory().NewExecutor()
+	organizations, err := usecases.Repositories.OrganizationRepository.AllOrganizations(ctx, exec)
+	if err != nil {
+		return err
+	}
+
+	for _, org := range organizations {
+		usecasesWithCreds := generateUsecaseWithCredForMarbleAdmin(ctx, usecases, org.Id)
+		dmUsecase := usecasesWithCreds.NewDataModelUseCase()
+		var dataModel models.DataModel
+		var err error
+		dataModel, err = dmUsecase.GetDataModel(ctx, org.Id)
+		if err != nil {
+			logger.ErrorContext(ctx, "Error getting data model for org "+org.Id, err)
+			return err
+		}
+		// iterate over tables
+		for _, table := range dataModel.Tables {
+			// 	start with the object_id field which needs to be unique in any case
+			field := table.Fields["object_id"]
+			if field.UnicityConstraint == models.NoUnicityConstraint {
+				fmt.Println()
+				logger.InfoContext(ctx, fmt.Sprintf("Adding index on object_id for table %s", string(table.Name)))
+				done, err := dmUsecase.AddUniqueIndex(
+					ctx,
+					org.Id,
+					table.Name,
+					field.Name,
+				)
+				if err != nil {
+					logger.ErrorContext(ctx, "Error adding index on object_id for table "+string(table.Name), err)
+					return err
+				} else if done {
+					logger.InfoContext(ctx, "Added index on object_id for table "+string(table.Name))
+				}
+				// refresh data model so that the field.UnicityConstraint is up to date for the next iteration
+				dataModel, err = dmUsecase.GetDataModel(ctx, org.Id)
+				if err != nil {
+					logger.ErrorContext(ctx, "Error getting data model for org "+org.Id, err)
+					return err
+				}
+			}
+
+		}
+		// iterate over tables again to treat link targets
+		for _, table := range dataModel.Tables {
+			// iterate over links to single and check the parent table fields
+			for _, link := range table.LinksToSingle {
+				parentTable := dataModel.Tables[link.LinkedTableName]
+				field := parentTable.Fields[link.ParentFieldName]
+				if field.UnicityConstraint == models.NoUnicityConstraint {
+					fmt.Println()
+					logger.InfoContext(ctx, fmt.Sprintf("Adding index on link target %s for table %s",
+						string(field.Name), string(parentTable.Name)))
+					done, err := dmUsecase.AddUniqueIndex(
+						ctx,
+						org.Id,
+						parentTable.Name,
+						field.Name,
+					)
+					if err != nil {
+						logger.ErrorContext(ctx, "Error adding index on object_id for table "+string(parentTable.Name), err)
+						return err
+					} else if done {
+						logger.InfoContext(ctx, "Added index on "+
+							string(field.Name)+" for table "+string(parentTable.Name))
+					}
+					// refresh data model so that the field.UnicityConstraint is up to date for the next iteration
+					dataModel, err = dmUsecase.GetDataModel(ctx, org.Id)
+					if err != nil {
+						logger.ErrorContext(ctx, "Error getting data model for org "+org.Id, err)
+						return err
+					}
```

Comment:
Here I understand the needs to do it inside the `for ... range` but this mind bug me ahah

Reply:
yes I understand 🙈

Reply:
anyway in this specific case it'll work: we're iterating over the original map, which we never update (we just replace the struct which contains a pointer to that map).
Because I just use it to get the fields, it's fine.
It's of course a bit fucked up, but this being a one-time script, I can live with it